modules/nextflow/src/main/groovy/nextflow/config/ConfigBuilder.groovy

```diff
@@ -735,5 +735,21 @@ class ConfigBuilder {
         buildConfigObject().toMap()
     }
 
+    /**
+     * Merge two maps recursively avoiding keys to be overwritten
+     *
+     * @param lmap
+     * @param rmap
+     * @return a map resulting of merging lmap and rmap
+     */
+    def mergeMaps(Map lmap, Map rmap) {
+        if (lmap.size() == 0) {
+            return rmap
+        }
+        rmap.each { k, v ->
+            lmap[k] = (lmap[k] in Map ? mergeMaps(lmap[k] as Map, v as Map) : v)
```

Comment:
I think it should be `lmap[k] instanceof Map && v instanceof Map` bit surprised, that works with `in`